web/frontend/src/lib/components/robot-client.svelte

```diff
@@ -430,12 +437,13 @@ const connect = async (creds?: Credentials) => {
 };
 
 const login = async (authType: string) => {
-  const creds = { type: authType, payload: password };
+  const creds = { type: authType, payload: passwordByAuthType[authType] ?? ""};
 
+  const authEntity = authType === apiKeyAuthType ? apiKeyEntity: undefined
   try {
-    await connect(creds);
+    await connect(creds, authEntity);
   } catch (error) {
-    notify.danger(`failed to connect: ${(error as ServiceError).message}`);
+    notify.danger(`failed to connect: ${JSON.stringify(error as ServiceError)}`);
```

Comment:
small drive-by: `.message` didnt exist always so it was showing up as "undefined" (very unhelpful error 🙃 )